pkg/tracker/gas.go

```diff
@@ -55,27 +64,27 @@ func (b *GasTracker) Exec(ctx context.Context) error {
 		if err != nil {
 			gasPrice, err = client.SuggestGasPrice(context.Background())
 			if err != nil {
-				log.Println("couldn't get suggested gas price", err)
+				level.Warn(b.logger).Log("msg", "couldn't get suggested gas price", "err", err)
```

Comment:
hm, I think for all trackers we want to return the error to the caller and the caller can decide whether to log it or pass it up the stack. This way we don't even need a logger inside the trackers.
```suggestion
				return errors.Wrap(err, "get suggested gas price")
```

Reply:
Won't we need the logger in case of Info and Debug messages anyway? 

If not, I don't mind removing the logger from trackers, but can we do it in a fresh PR?

Reply:
Yes you are right I just double-checked and some are printing some debug logs.